### PR TITLE
Fix `from_dataset` constructor in `KeyObjectSelectionDocument`

### PR DESCRIPTION
**Pull request by cgorman**
Fixed the class method `KeyObjectSelection.from_dataset()` and added a test for it.

I had to change a bit around so you may want to pay close attention to what I did to ensure it makes sense.

### CODE REVIEW

File: src/highdicom/ko/content.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def from_sequence(
                 'Item #1 of sequence is not an appropriate SR Content Item '
                 'because it does not have Template Identifier "2010".'
             )
-        instance = ContentSequence.from_sequence(sequence, is_root=True)
+        instance = ContentSequence.from_sequence(sequence, is_root=is_root)
```
Comment (hackermd) [inline src/highdicom/ko/content.py:175]: I would keep this hardcoded. We just keep the `is_root` parameter to satisfy the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)

File: tests/test_ko.py [hunk 2]
```diff
@@ -201,3 +201,37 @@ def test_construction(self):
...
+            sop_instance_uid=UID(),
+            instance_number=1,
+            manufacturer='MGH Computational Pathology',
+            institution_name='Massachusetts General Hospital',
+            institutional_department_name='Pathology'
+        )
+        assert isinstance(document, KeyObjectSelectionDocument)
+        assert isinstance(document.content, KeyObjectSelection)
+
+        test_document = KeyObjectSelectionDocument.from_dataset(document)
```
Comment (hackermd) [inline tests/test_ko.py:220]: That is maybe a bit too simplistic. I would suggest to serialize the `document` instance to a buffer and read it back out using the pydicom `dcmwrite()` and `dcmread()` methods, respectively (see `test_seg.py` for examples).

**Review by hackermd — CHANGES_REQUESTED** [review]

File: src/highdicom/ko/content.py (hunk 1 above)
Comment (cgorman) [reply src/highdicom/ko/content.py]: Fixed in `0b52144`

File: tests/test_ko.py (hunk 2 above)
Comment (cgorman) [reply tests/test_ko.py]: Fixed in `0b52144`

**Review by hackermd — APPROVED** [review]

**Pull request merged**